### PR TITLE
My hw 04

### PR DESCRIPTION
**Pull request by Ikl0**

### CODE REVIEW

File: HW04/ivan_klimov/lib/mentor.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Mentor < Person
+  def add_homework(title:, description:, student:)
+    hw = Homework.new(title: title, description: description, student: student)
```
Comment (B0RlS) [inline HW04/ivan_klimov/lib/mentor.rb:5]: add initializer pls

File: HW04/ivan_klimov/lib/notification.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Notification
+  def nf_is_empty(person)
```
Comment (B0RlS) [inline HW04/ivan_klimov/lib/notification.rb:4]: could you please add full name to all methods

File: HW04/ivan_klimov/lib/file_module.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module LogSystem
+  def create_log_file(person:)
```
[the remark this answers is not in the document]
Comment (Ikl0) [reply HW04/ivan_klimov/lib/file_module.rb]: This is file is not needed in the current version of the program, so I removed it

File: HW04/ivan_klimov/lib/mentor.rb (hunk 1 above)
Comment (Ikl0) [reply HW04/ivan_klimov/lib/mentor.rb:5]: I have a parent class Person, with initializer. So child classes Mentor and Student are inherited from it

Comment (vladhilko) [inline HW04/ivan_klimov/lib/mentor.rb:5]: Let's use full names everywhere
```suggestion
    homework = Homework.new(title: title, description: description, student: student)
```

File: HW04/ivan_klimov/lib/notification.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Notification
+  def check_for_active_notifications(person)
```
Comment (vladhilko) [inline HW04/ivan_klimov/lib/notification.rb:4]: we usually use the following names 
```suggestion
  def active_notifications_present?(person)
```

File: HW04/ivan_klimov/lib/notification.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    person.notifications.include?("#{person.name}, you have read all notifications.")
+  end
+
+  def push_homework(title)
+    " '#{Time.new.strftime('%d-%m-%Y %H:%M:%S')}' new homework : #{title}"
```
Comment (vladhilko) [inline HW04/ivan_klimov/lib/notification.rb:9]: you use the same method many times, let's encapsulate it to something like this

```rb
def current_time
  Time.new.strftime('%d-%m-%Y %H:%M:%S')
end
```

**Pull request merged**